src/api/tests/test_user_get_and_put.py

```diff
@@ -0,0 +1,111 @@
+from urllib import response
+from .util import Client
+
+
+TEST_USER = { 'email': "test@email.com",
+              'password': "123456" }
+
+
+
+TEST_USER_SIGNUP = { 'email': 'test@email.com',
+                     'name': 'TestName',
+                     'phone': '',
+                     'password': '123456',
+                     'degree': 'Undergraduate',
+                     'major': 'CSCI' }
+
+
+TEST_USER2 = { 'email': "test2@email.com",
+              'password': "1234567" }
+
+
+TEST_USER_SIGNUP2 = { 'email': 'test2@email.com',
+                     'name': 'TestName2',
+                     'phone': '',
+                     'newPassword': '1234567',
+                     'degree': 'Graduate',
+                     'major': 'ECON' }
+
+
+'''
+Test this api endpoint/file only with the following command line:
+pytest -s tests/test_user_course.py
+'''
+
+def test_get_user_success(client: Client):
+    '''
+    Test user get by using /api/session and TEST_USER_SIGNUP.
+    '''
+    r = client.post("/api/session", json=TEST_USER)
+    assert r.status_code == 200
+    data = r.json()
+    assert data['content'] is not None
+    assert data['content']['sessionID'] is not None
+    assert data['content']['userName'] == TEST_USER_SIGNUP['name']
+    r = client.get("/api/user/"+data['content']['sessionID'])
+    assert r.status_code == 200
+    data = r.json()
+    assert data['content']['degree'] == TEST_USER_SIGNUP['degree']
+    assert data['content']['email'] == TEST_USER_SIGNUP['email']
+    assert data['content']['major'] == TEST_USER_SIGNUP['major']
+    assert data['content']['name'] == TEST_USER_SIGNUP['name']
+    assert data['content']['phone'] == TEST_USER_SIGNUP['phone']
+    assert data['content']['uid'] is not None
+
+
+def test_get_user_failed(client: Client):
```

Comment:
I've just noticed the param session id in the API request was only used to get the user from the database, not for verifying the user is in session. Keep this test case but add another with the following steps:
1. Create a new session and save the session id
2. Close the session
3. call get /api/session/<session id from step 1>